python/cudf/cudf/core/column/column.py

```diff
@@ -179,11 +179,9 @@ def any(self):
         return bool(libcudf.reduce.reduce("any", self, dtype=np.bool_))
 
     def __sizeof__(self):
-        n = self.base_data.size if self.base_data is not None else 0
+        n = self.data.size
         if self.nullable:
-            n += self.base_mask.size
-        for child in self.base_children:
-            n += child.__sizeof__()
+            n += self.base_mask.size if self.offset == 0 else self.mask.size
```

Comment:
We shouldn't use `self.mask` as that can copy construct a new mask buffer in the case of a non-zero offset. Instead we could use the utility function that we have to determine the size the copy constructed mask would be and return that.